src/Microsoft.Sbom.Api/Workflows/Helpers/RelationshipsArrayGenerator.cs

```diff
@@ -26,92 +26,92 @@ public class RelationshipsArrayGenerator : IJsonArrayGenerator<RelationshipsArra
 
     private readonly ILogger log;
 
-    private readonly ISbomConfigProvider sbomConfigs;
-
     private readonly IRecorder recorder;
 
+    public ISbomConfig SbomConfig { get; set; }
+
+    public string SpdxManifestVersion { get; set; }
+
     public RelationshipsArrayGenerator(
         RelationshipGenerator generator,
         ChannelUtils channelUtils,
         ILogger log,
-        ISbomConfigProvider sbomConfigs,
         IRecorder recorder)
     {
         this.generator = generator;
         this.channelUtils = channelUtils;
         this.log = log;
-        this.sbomConfigs = sbomConfigs;
         this.recorder = recorder;
     }
 
-    public async Task<IList<FileValidationResult>> GenerateAsync()
+    public async Task<GenerateResult> GenerateAsync()
     {
         using (recorder.TraceEvent(Events.RelationshipsGeneration))
         {
-            IList<FileValidationResult> totalErrors = new List<FileValidationResult>();
+            var totalErrors = new List<FileValidationResult>();
+            var serializersToJsonDocs = new Dictionary<IManifestToolJsonSerializer, List<JsonDocument>>();
```

Comment:
This pattern of Dictionary to a list of documents is showing up a lot. Maybe we should consider a templated class to wrap the type and adding to a new document? Something like:
```
internal class IndexedListCollection<T>
{
  internal IDictionary<T, List<JsonDocument>> Dict { get; }

  internal void AddDocumentToCollection(T key, JsonDocument document)
  {
    if (Dict.TryGetValue(key, out documentList))
    {
      documentList.Add(document);
      return;
    }

    Dict.Add(key, new List<JsonDocument>{ document });
  }
}
```

Reply:
Done, this does clean up the code significantly thank you for the suggestion!